### PR TITLE
Remove friendly_id

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review.

File: backend/app/views/spree/admin/products/new.html.erb [hunk 1]
```diff
@@ -14,6 +14,12 @@
       <%= f.error_message_on :name %>
     <% end %>
 
+    <%= f.field_container :slug do %>
+      <%= f.label :slug, Spree.t(:slug) %> <span class="required">*</span><br />
+      <%= f.text_field :slug, :class => 'fullwidth title', :required => true %>
```
Comment (dkubb) [inline backend/app/views/spree/admin/products/new.html.erb]: This should use 1.9 style hashes.

File: core/app/models/spree/product.rb [hunk 2]
```diff
@@ -85,19 +84,20 @@ class Product < Spree::Base
...
+    before_validation :find_or_build_master
     before_validation :validate_master
 
     validates :name, presence: true
     validates :shipping_category_id, presence: true
-    validates :slug, length: { minimum: 3 }, uniqueness: { allow_blank: true }
+
+    validates :slug, presence:   true,
+                     length:     { allow_blank: true, minimum: 3                                       },
+                     format:     { allow_blank: true, unless: :deleted_at, with: /\A#{SLUG_REGEXP}\z/, },
```
Comment (dkubb) [inline core/app/models/spree/product.rb]: The hash has a trailing comma.

File: core/app/models/spree/product.rb [hunk 3]
```diff
@@ -85,19 +84,20 @@ class Product < Spree::Base
...
-    before_validation :normalize_slug, on: :update
+    before_validation :find_or_build_master
     before_validation :validate_master
 
     validates :name, presence: true
     validates :shipping_category_id, presence: true
-    validates :slug, length: { minimum: 3 }, uniqueness: { allow_blank: true }
+
+    validates :slug, presence:   true,
+                     length:     { allow_blank: true, minimum: 3                                       },
```
Comment (dkubb) [inline core/app/models/spree/product.rb]: Should we specify the maximum length of 255 to match the column limit?

Comment (mbj) [inline core/app/models/spree/product.rb]: yeah, good catch.

**Comment by mbj** [comment]
@dkubb Ready for the next pass.

**Pull request merged**